components/datetime/src/neo_marker.rs

```diff
@@ -1542,308 +1545,507 @@ macro_rules! datetime_marker_helper {
     (@names/zone/specific_short, yes) => {
         tz::MzSpecificShortV1Marker
     };
-    (@names/$any:ident, no) => {
+    (@names/$any:ident,) => {
+        NeverMarker<()>
+    };
+    (@names/$any:ident,) => {
         NeverMarker<()>
     };
-    (@names/zone/$any:ident, no) => {
+    (@names/zone/$any:ident,) => {
         NeverMarker<()>
     };
+    () => {
+        unreachable!() // prevent bugs
+    };
+}
+
+/// Maps the token `yes` to the given ident
+macro_rules! yes_to {
+    ($any:ident, yes) => {
+        $any
+    };
+    () => {
+        unreachable!() // prevent bugs
+    };
 }
 
 /// Generates the options argument passed into the docs test constructor
 macro_rules! length_option_helper {
-    (no) => {
+    () => {
         stringify!(Default::default())
     };
     ($length:ident) => {
         concat!("NeoSkeletonLength::", stringify!($length), ".into()")
     };
 }
 
+/// Generates an impl of [`NeoGetField`]
+macro_rules! impl_get_field {
+    ($type:path, never) => {
+        impl NeoGetField<NeverField> for $type {
```

Comment:
Yes, it's using the same trait because the signature is what I need and it is reasonably close semantically. I am happy to rename the trait. As noted previously, I see the trait like `AsRef` except that it returns by value instead of by reference.

@Manishearth